oktapam/resource_database.go

```diff
@@ -0,0 +1,420 @@
+	canonicalName := d.Get(attributes.CanonicalName).(string)
+	dbType := d.Get(attributes.DatabaseType).(string)
+
+	var selectorLabels *map[string]string
+	if selector, ok := d.GetOk(attributes.ManagementGatewaySelector); ok {
+		if s, ok := selector.(map[string]any); ok {
+			tmpLabels := make(map[string]string, len(s))
+			for k, v := range s {
+				if value, ok := v.(string); ok {
+					tmpLabels[k] = value
+				}
+			}
+			selectorLabels = &tmpLabels
+		} else {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "invalid label selector",
+			})
+		}
+	}
+
+	mgmtType, mgmtDetails, err := mgmtConnectionDetailsFromResource(ctx, c, d)
+	if err != nil {
+		return err
+	}
+
+	if dbID == "" {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "could not obtain database id from resource",
+		})
+	}
+	if resourceGroupID == "" {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "could not obtain resource group id from resource",
+		})
+	}
+	if projectID == "" {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "could not obtain project id from resource",
+		})
+	}
+
+	if diags != nil {
+		return diags
+	}
+
+	if err := client.UpdateDatabase(ctx, c, dbID, resourceGroupID, projectID, canonicalName, dbType, mgmtType, *mgmtDetails, selectorLabels); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceDatabaseRead(ctx, d, m)
+}
+
+func resourceDatabaseDelete(ctx context.Context, d *schema.ResourceData, m any) diag.Diagnostics {
+	c := getSDKClientFromMetadata(m)
+	resourceGroupID := d.Get(attributes.ResourceGroup).(string)
+	projectID := d.Get(attributes.Project).(string)
+	databaseID := d.Get(attributes.ID).(string)
+
+	err := client.DeleteDatabase(ctx, c, resourceGroupID, projectID, databaseID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceDatabaseReadImport(ctx context.Context, d *schema.ResourceData, m any) ([]*schema.ResourceData, error) {
+	resourceGroupID, projectID, databaseID, err := parseDatabaseID(d.Id())
+	if err != nil {
+		return nil, err
+	}
+	if err := d.Set(attributes.ResourceGroup, resourceGroupID); err != nil {
+		return nil, err
+	}
+	if err := d.Set(attributes.Project, projectID); err != nil {
+		return nil, err
+	}
+	d.SetId(databaseID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func parseDatabaseID(resourceId string) (string, string, string, error) {
+	split := strings.Split(resourceId, "/")
+	if len(split) != 3 {
+		return "", "", "", fmt.Errorf("expected format: <resource_group_id>/<project_id>/<id>, received: %s", resourceId)
+	}
+	return split[0], split[1], split[2], nil
+}
+
+func mgmtConnectionDetailsFromResource(_ context.Context, pamClient client.SDKClientWrapper, d *schema.ResourceData) (string, *pam.ManagementConnectionDetails, diag.Diagnostics) {
+	v, ok := d.GetOk(attributes.ManagementConnectionDetails)
```

Comment:
You don't need these checks. It's already part of schema.

Reply:
i was starting to realize we can trust the schema for more of this.  will have another scan through and clean things up

Reply:
Removed in 3608574a504bb4213ac00b172682c23fe98226dc